### PR TITLE
Fix set-cookie sent multiple times when regenerating session

### PR DESCRIPTION
**Pull request by matthistuff**
bildschirmfoto 2014-02-27 um 13 17 22
When logging in (account/login), Shopware somtimes will return a Set-Cokie header for the same cookie name three times (old sessionID, new sessionID, new sessionID). This might lead to unexpected behavior, as it is not clear what happens in the browser when setting the same cookie multiple times.

This is not the fix for the root problem of, but mentioned in a sidenote here: http://jira.shopware.de/?ticket=SW-7684

### CODE REVIEW

File: engine/core/class/sAdmin.php [hunk 1]
```diff
@@ -1296,6 +1296,8 @@ public function regenerateSessionId()
         session_regenerate_id(true);
         $newSessionId = session_id();
 
+        // clear any set-cokie header before opening a new session
+        header_remove('Set-Cookie');
```
Comment (bcremer) [inline engine/core/class/sAdmin.php:1300]: But this will also remove any other cookie that may be not session related?

Comment (matthistuff) [inline engine/core/class/sAdmin.php:1300]: That is indeed true. Is there a better way for just unsetting the session-x cookies?

Comment (bcremer) [inline engine/core/class/sAdmin.php:1300]: We have a ticket open for the 4.2.2 release and will further investigate into this issue.

**Comment by dnoegel** [comment]
Hi,

we've a ticket for that here: jira.shopware.de/?ticket=SW-8413

We will further investigate on this and close here.

Thanks for your contribution.

**Pull request closed without merging**